import { TransformError } from './transformError.js'

describe('TransformError', () => {
  test('the error is formatted correctly', async () => {
    const error = new TransformError(
      {
        schema: 'public',
        table: 'User',
        columns: ['id', 'name', 'email'],
        row: {
          line: 3654,
          raw: {
            id: 'u1654519848516',
            name: 'John Doe',
            email: 'john.doe@gmail.com',
          },
          parsed: {
            id: 'u1654519848516',
            name: 'John Doe',
            email: 'john.doe@gmail.com',
          },
        },
        column: 'name',
      },
      'it broke here'
    )

    expect(error.toString()).toMatchInlineSnapshot(`
      "       â”Œâ”€[\\"public\\".\\"User\\"]
             â”‚
           1 â”‚  id,name,email
             Â·
             Â·
      [31mâˆ™[39m 3655 â”‚  u1654519848516,John Doe,john.doe@gmail.com
             Â·  [31m               â”€â”€â”€â”€â”¬â”€â”€â”€[39m
             Â·  [31m                   â•°â”€â”€â”€â”€â”€â”€ it broke here[39m
             â”‚
             â””â”€"
    `)
  })
})
